### DISCUSSION TITLE
Issue with Bridge

### DISCUSSION THREAD

**Discussion by equalogix** [zkSync-Community-Hub/zksync-developers#23] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I tried to bridge ETH to zkSync-Ere ETH using the zkSync bridge. The transaction got stuck and remains stuck and I have no way resolving it without adding ETH to my account. I think I understand what the problem is, I think it is a UI issue with the "Max" feature. When setting the amount to bridge I used the "Max" button, the system then populated the transaction by deducting the estimated gas from the total balance in my account and then used the remaining balance as the amount to bridge. This is logically what using "Max" suggests. Now due to fee market volatility, the estimated fee was not sufficient to get the transaction included into the block, thus the transaction remains pending. The problem is that all the funds in the account are either included in transaction as fees or as amount to bridge effectively leaving the balance in the account at 0. Thus making it impossible to either speed-up or cancel the transaction, as both these actions require additional funds in the account. The only solution is to wait for the fee market to come back down, or to add funds to the account.

This is an edge case for sure, I intentionally executed the transaction at night when fees would be low. My suggestion would be that when setting the "Max", that a small amount be left in the wallet, probably 15% or 25% of the estimated gas fee to allow the user the possibility to "speed-up" the transaction if fails to get included in the block. And/Or a warning that there is a risk that the transaction could stall.

### What is the specific issue or error you're encountering?

Not applicable

### Can you share the error messages or logs you're receiving, if any?

Not applicable

### Have you made any recent changes to the contract before encountering this issue?

Not applicable

### Are there any external libraries or contracts that your contract interacts with?

Not applicable

### Can you provide the relevant portions of your contract code where the issue is occurring?

Not applicable

### Have you tried to isolate the problem, and if so, what were the results?

Not applicable

### What steps have you already taken to try to resolve the issue?

Not applicable

### Additional Details

Note:
- I realize this may not be the best place to raise this issue, but I couldn't think of a better place, so feel free to move it.
- I'm @BlindChicken on discord.

**Comment by equalogix** [comment, marked as answer]
I solved my problem, and it does not seem to be anything to do with what I described above. Actually I'm not sure what the issue is. I was able to cancel the transaction by sending myself 0 ETH with the same nonce and boosted gas fees. I then went back to the bridge and tried again this time without using the "Max" and the transaction got stuck again. Once again in Metamask I tried speeding up the transaction and at each attempt it failed again with a `Replacement Transaction UnderPriced` Error. After Googling around a bit I found a Youtube video that suggested the solution to the problem was increasing the `Priority Fee`. Well that worked!

Why do the transaction from zkSync Bridge get stuck, I don't know!
But if the transaction does get stuck, crank the priority fee in Metamask.

**Comment by newtondav** [comment]
Thank you for bringing this issue to our attention. Your suggestion to leave a small amount in the wallet when using the "Max" feature to bridge funds is valuable and could help prevent similar situations in the future. We'll consider implementing this feature or providing a warning about the potential risk of transaction stalling in such cases. Your feedback is appreciated!

**Comment by fullertasha** [comment]
Suggestion: 
Consider modifying the "Max" feature to leave a portion of the balance in the wallet to cover potential fee fluctuations, avoiding transaction stalls.

**Comment by unticcutecerin** [comment]
The user encountered an issue when bridging ETH to zkSync-Ere ETH using the zkSync bridge. They used the "Max" button to set the amount to bridge, which deducted the estimated gas from their total balance and used the remaining balance as the amount to bridge. Due to fee market volatility, the estimated fee was not sufficient to get the transaction included in the block, leaving their account balance at 0 and making it impossible to speed up or cancel the transaction. The user suggests leaving a small amount in the wallet when using "Max" or providing a warning about the risk of stalling the transaction. Without further information or access to the code repository, it is difficult to provide a specific solution. However, it may be helpful for the user to reach out to the zkSync support team for assistance in resolving the issue.
